consensus/src/util/time_service.rs

```diff
@@ -135,13 +136,13 @@ pub enum WaitingSuccess {
 }
 
 /// Error states for wait_if_possible
-#[derive(Debug, PartialEq, Eq, Fail)]
+#[derive(Debug, PartialEq, Eq, Error)]
 pub enum WaitingError {
     /// The waiting period exceeds the maximum allowed duration, returning immediately
-    #[fail(display = "MaxWaitExceeded")]
+    #[error("MaxWaitExceeded")]
```

Comment:
Good point, thanks for point this out since i'm not super familiar with this section of code i'll probably use the `#[error("{:?}", self)]` trick you pointed out below and let the folks who know this better work to improve the error message.